src/main/io/serial.c

```diff
@@ -572,41 +572,49 @@ void closeSerialPort(serialPort_t *serialPort)
     serialPortUsage->serialPort = NULL;
 }
 
-void serialInit(bool softserialEnabled, serialPortIdentifier_e serialPortToDisable)
+void serialInit(bool softserialEnabled, bool serialAvoidanceCheck)
 {
-#if !defined(USE_SOFTSERIAL)
-    UNUSED(softserialEnabled);
+    memset(&serialPortUsageList, 0, sizeof(serialPortUsageList));
+
+/*
+    TODO: validate this avoidance is still necessary
```

Comment:
No config uses AVOID_UART[1-3]_FOR_PWM_PPM. IIRC it was used on F1 and nobody was brave enough to delete it. 

PPM/PWM receiver is IMO broken for some time, if necessary, some test may be implemented when resurrecting it.

BTW: Maybe it is time to remove PWM/PPM input completely.